app/src/main/java/de/danoeh/antennapod/activity/VideoplayerActivity.java

```diff
@@ -567,7 +568,14 @@ public boolean onPrepareOptionsMenu(Menu menu) {
         menu.findItem(R.id.disable_sleeptimer_item).setVisible(controller.sleepTimerActive());
 
         menu.findItem(R.id.player_switch_to_audio_only).setVisible(true);
-        menu.findItem(R.id.audio_controls).setIcon(R.drawable.ic_sliders);
+
+        List<String> audioTracks = controller.getAudioTracks();
```

Comment:
Hmm, I'm not 100% sure but I think this might block the UI thread if the controller is not ready yet. If it doesn't block it, it might return an empty list while the media file is not parsed yet. Then it hides the menu item even if the media file later turns out to have audio tracks. I think it would be better to handle the show/hide decision in `newPlaybackController.loadMediaInfo` or (if the proper location does not work) in `newPlaybackController.updatePlayButtonShowsPlay`.

You can find a podcast with multiple audio tracks for testing here: https://media.ccc.de/c/37c3/podcast/mp4-lq.xml

Reply:
Alternatively, we could just always show it like it currently is (and maybe add a message saying "episode does not have multiple audio tracks" or something)

Reply:
Thanks, I was struggling to find one with multiple audio tracks.
I guess the standard thing I would expect is to always see the menu option, and for the list of tracks to include the currently selected one. Seems like currently it doesn't show the currently selected one, but if it did show it that would also avoid the possibility of a blank screen.

Reply:
I've taken a look and I see that the way it works is it shows a single button with the current track, then tapping it goes to the next track so my suggestion above would involve changing how this interface works.

Reply:
I guess another alternative would be to show the audio track name as text rather than a button when there's only 1 track.

Reply:
I'm not entirely sure what this is about, but from what I can read: 80% of the episodes probably only have 1 audio track. In that case I think it's better (because simpler UI for most users) if we don't display anything.

Reply:
Yeah, showing/hiding the menu option is probably the best thing to do, but it needs some more work. To keep the PR more focused on a single task, I would just keep the dialog in the overflow menu for now. It only applies to video anyway, so it is probably not something average users stumble upon that often.